src/zoom.ts

```diff
@@ -1,257 +1,213 @@
-import request from 'request-promise';
-import { ResponseError } from './webhooks/helpers';
+import request from "request-promise";
+import { ResponseError } from "./webhooks/helpers";
 import {
-	ZOOM_MEETING_ID,
-	getZoomToken,
-	ZOOM_TIME_THRESHOLD,
-	ZOOM_TIME_ANNOUNCEMENT_CHANNEL,
-	ZOOM_TIME_DEBOUNCE_HOURS,
-} from './config';
-import { Client, TextChannel } from 'discord.js';
-import { formatMessage } from './helpers';
-
-export interface ZoomInfo {
-	meetingInfo: any;
-	active: boolean;
-	hasSeenStart: boolean;
-	participants: ZoomParticipant[];
-}
-
-interface ZoomParticipant {
-	id: string;
-	name: string;
-}
+  ZOOM_MEETING_ID,
+  getZoomToken,
+  ZOOM_TIME_THRESHOLD,
+  ZOOM_TIME_ANNOUNCEMENT_CHANNEL,
+  ZOOM_TIME_DEBOUNCE_HOURS,
+  PRODUCTION,
+} from "./config";
+import { dispatch, getState } from "./store";
+import {
+  getOnlineUsers,
+  getIsActive,
+  userLeft,
+  userJoined,
+  callStarted,
+  callEnded,
+  setJoinUrl,
+} from "./store/zoom";
 
-let hasSeenStart: boolean = false;
-let participants: ZoomParticipant[] = [];
 let lastZoomTime: number = -1; // Date.now() format
 
 /**
  * Check the status of the Zoom call.
  */
-export async function getZoomInfo(): Promise<ZoomInfo> {
-	const meetingInfo = await request.get({
-		uri: `https://api.zoom.us/v2/meetings/${ZOOM_MEETING_ID}`,
-		auth: {
-			bearer: getZoomToken(),
-		},
-		headers: {
-			'User-Agent': 'Hanbot',
-			'Content-Type': 'application/json',
-		},
-		json: true,
-	});
-
-	const active = meetingInfo.status === 'started';
-
-	if (!active) {
-		hasSeenStart = true;
-		participants = [];
-	}
-
-	if (!hasSeenStart) participants = [];
-
-	return { meetingInfo, active, hasSeenStart, participants };
-}
-
-export async function updateDiscordStatusFromZoom(discord: Client, zoomInfo?: ZoomInfo) {
-	if (!zoomInfo) zoomInfo = await getZoomInfo();
-
-	if (zoomInfo.active) {
-		// Curently, the outer ternary is redundant because status is only set when zoomInfo.active is true
-		const status = zoomInfo.active
-			? hasSeenStart && participants.length >= 0
-				? `with ${
-						participants.length === 1 ? `1 person` : `${participants.length} people`
-				  } on Zoom`
-				: 'on Zoom'
-			: 'with nobody'; /* currently disabled */
-
-		console.log(`Setting Discord status to: ${status}`);
-		await discord.user?.setActivity(status, { type: 'PLAYING' });
-	} else {
-		console.log('Clearing Discord Status');
-		// This is the best way I can find to clear activity (bots cant have custom statuses)
-		await discord.user?.setActivity('', { type: 'CUSTOM_STATUS' });
-	}
+export async function updateZoomStatus(): Promise<void> {
```

Comment:
I think this should be converted to a thunk that must be dispatched for consistency, because right now it looks like it _should_ need to be dispatched, but magically doesn't.

Reply:
I get what you're saying, but am strongly opposed to introducing thunks.
IMO The way that they overload dispatch adds a ton of mental overhead (especially in TS) and makes your code a lot less predictable (you are no longer able to look at a call to dispatch and your reducers and know how it will change the state).
Anecdotally, I've found thunks are really hard for a lot of people to pick up, which could be an issue if anyone else want's to contribute.
While are passable for representing super simple async stuff (like fetch a resource and dispatch an action with the complete response), they become a PITA to debug & grok once you have any sort of conditional dispatches going on.

Reply:
Good point, especially on the new people. Let's keep it this way.